### DISCUSSION TITLE
Supabase & NextJS

### DISCUSSION THREAD

**Discussion by YummyPampers2** [supabase/supabase#1911] [Questions]
I am attempting to use Supbase as the backend for my next-auth app. 

In my [nextauth].js file I am using the following database function + parameters
for local tests:

```
database: {
        type: "supabase",
        database: ":memory:",
        synchronize: true,
    }
```

But am returning the following error:

```
npm ERR! 404  'create-supabase@latest' is not in the npm registry

You should bug the author to publish or name it yourself 
```

I suppose posting here would be considered bugging the author? Otherwise, 
might anyone know if this module goes by another name? Or how to begin renaming this module? 

Thanks,

**Comment by awalias** [comment]
Hey @YummyPampers2 it looks like your app is trying to import 'create-supabase@latest' which seems wrong

I don't have a next-auth example handy but I do have a "Next Auth" example :P which might give some hints...

https://github.com/supabase/supabase/tree/master/examples/nextjs-auth

**Reply by YummyPampers2** [reply]
Hi Ant: 

Thank you for sharing, I think the direction I wanted 
to go, is to create an Adapter. From the NextJS site 
the over presents how to create a custom adapter 
[HERE](https://next-auth.js.org/tutorials/creating-a-database-adapter)

Goal: 
Use Supabase instead of SQLite for my NextAuth.js  backend. 

Steps:
First The npm dependencies I have imported are: 
```
>> @supabase/supabase-js@1.13.2
>> supabase@0.4.2
>> next-auth
// want to replace firebase with supabase
>> [attempting] @next-auth/firebase-adapter@canary
```
Then -- I am attempting to create one similar to the firebase method
in my [...nextauth].js file as: 

```
import { FirebaseAdapter } from "@next-auth/firebase-adapter"

import firebase from "firebase/app"
import "firebase/firestore"

const firestore = (
  firebase.apps[0] ?? firebase.initializeApp(/* your config */)
).firestore()


export default NextAuth({
  providers: [
    Providers.Google({
      clientId: process.env.GOOGLE_ID,
      clientSecret: process.env.GOOGLE_SECRET,
    }),
  ],
  adapter: FirebaseAdapter(firestore),
  ...
})

``` 

Questions: 
1. Is there a similar method using supabase?
2. How should I configure the supbase function?
